### PR TITLE
feature: linked scan activity to book adapters

### PR DESCRIPTION
**Pull request by shadowciaw**
#### Summary of changes/features added:
- clicking borrow/lend/return buttons now open scan activity to scan barcode

#### Screenshots of UI changes (update with each new commit): 
- you click on the buttons above and it brings you to scan activity

## Pull Request Checklist
- [ ] Unit test written?
- [ ] (**For later in project**) Integration tests written?
- [x] Java docs written?

### CODE REVIEW

File: client/app/src/main/java/com/example/criengine/Adapters/BorrowerBooksListAdapter.java [hunk 1]
```diff
@@ -29,25 +34,44 @@
...
+    }
+
     /**
      * Returns a view with the properly formatted information.
-     * @param position The position from the list.
+     *
+     * @param position    The position from the list.
      * @param convertView The old view to reuse (if possible).
-     * @param parent The parent view group.
+     * @param parent      The parent view group.
```
Comment (AdityaHarvi) [inline client/app/src/main/java/com/example/criengine/Adapters/BorrowerBooksListAdapter.java:74]: Does this affect java docs? The spacing?

File: client/app/src/main/java/com/example/criengine/Adapters/BorrowerBooksListAdapter.java [hunk 2]
```diff
@@ -82,8 +106,8 @@ public void onClick(View v) {
 
         statusText.setText(book.getStatus());
 
-        actionButton.setEnabled(true);
         actionButton.setVisibility(View.VISIBLE);
+        actionButton.setEnabled(true);
         switch (book.getStatus()) {
             case "borrowed":
 //            statusText.setTextColor(view.getResources().getColor(R.color.status_accepted));
```
Comment (AdityaHarvi) [inline client/app/src/main/java/com/example/criengine/Adapters/BorrowerBooksListAdapter.java:113]: We might be able to just remove this actually, i don't think we're worrying about coloring for the text. Lets us clean up the file a tiny bit :D

File: client/app/src/main/java/com/example/criengine/Database/DatabaseWrapper.java [hunk 3]
```diff
@@ -502,10 +502,10 @@ public Boolean apply(@NonNull Transaction transaction) throws FirebaseFirestoreE
     }
     public Task<Boolean> borrowBook (final String bookID, final String ISBN) {
         return db.runTransaction(new Transaction.Function<Boolean>() {
-
             @Nullable
             @Override
             public Boolean apply(@NonNull Transaction transaction) throws FirebaseFirestoreException {
+                Log.d("testingDB", bookID);
```
Comment (AdityaHarvi) [inline client/app/src/main/java/com/example/criengine/Database/DatabaseWrapper.java:508]: do we want this to be in the final version?

File: client/app/src/main/java/com/example/criengine/Fragments/RequestedBooksFragment.java [hunk 4]
```diff
@@ -123,4 +130,26 @@ public void onConfirmPressed(List<String> activeFilters) {
...
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+
+        // Check that it is the ScanActivity with an OK result
+        if (requestCode == SCAN_RESULT_CODE) {
+            if (resultCode == RESULT_OK) {
+                // Get String data from Intent
+                String barcodeData = data.getStringExtra("barcode");
+                String bookID = data.getStringExtra("bookID");
+                Log.d("testing", "barcode data=" + barcodeData);
```
Comment (AdityaHarvi) [inline client/app/src/main/java/com/example/criengine/Fragments/RequestedBooksFragment.java:149]: another log O:

**Review by SamLojpur — APPROVED** [review]
Made some tweaks, works with the DB now!

**Pull request merged**